### ISSUE TITLE
Asset creation fee issue

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#433]
When creating an asset, if the fee is not paid in the CORE asset but in another asset, an amount of extra CORE asset will be created from nowhere unexpectedly.

The workflow is `prepare_fee` -> `do_evaluate` -> `convert_fee` -> `pay_fee` -> `do_apply`.

* `prepare_fee` is in [evaluator.cpp](https://github.com/bitshares/bitshares-core/blob/2.0.170710/libraries/chain/evaluator.cpp#L76), `core_fee_paid` got initialized normally:
```
         core_fee_paid = fee_from_pool.amount;
```

* `do_evaluate` is in [asset_evaluator.cpp](https://github.com/bitshares/bitshares-core/blob/2.0.170710/libraries/chain/asset_evaluator.cpp#L94),  `core_fee_paid` is halved:
```
   core_fee_paid -= core_fee_paid.value/2;
```

* `convert_fee` is in [evaluator.cpp](https://github.com/bitshares/bitshares-core/blob/2.0.170710/libraries/chain/evaluator.cpp#L89), the halved `core_fee_paid` is deducted from the fee paying asset's fee pool (**this is incorrect, instead, we should deduct full amount from the fee pool**):
```
               d.fee_pool -= core_fee_paid;
```

* `pay_fee` is in [evaluator.cpp](https://github.com/bitshares/bitshares-core/blob/2.0.170710/libraries/chain/evaluator.cpp#L102), the halved `core_fee_paid` is sent to referral program:
```
            s.pay_fee( core_fee_paid, d.get_global_properties().parameters.cashback_vesting_threshold );
```

* `do_appy` is in [asset_evaluator.cpp](https://github.com/bitshares/bitshares-core/blob/2.0.170710/libraries/chain/asset_evaluator.cpp#L127), the halved `core_fee_paid` got added to the new asset's fee pool**:
```
         a.fee_pool = core_fee_paid; //op.calculate_fee(db().current_fee_schedule()).value / 2;
```

A soft fork is proposed to prevent this issue from being exploited: #432.

**Comment by btsfav** [comment]
Is there an ETA? crypto-bridge is already complaining, since they rely on creating assets

**Comment by CryptoBridge** [comment]
Yes, an ETA would be good. This is already damaging to us as we have clients already asking us what's up and why their coins can't be added to the exchange. Thanks.

**Comment by abitmore** [comment]
@btsfav @CryptoBridge when majority of witnesses has applied the soft fork, the committee can reduce the fee. Hopefully will be done in 24 hours.

**Comment by abitmore** [comment]
Should be fixed in release `2.0.171025`. Closing.
